rtgui/batchqueue.cc

```diff
@@ -928,8 +953,11 @@ Glib::ustring BatchQueue::calcAutoFileNameBase (const Glib::ustring& origFileNam
                 } else if (options.savePathTemplate[ix] == 'p') {
                     // insert path elements from the start of the path up to the given index
                     ix++;
-                    unsigned int n = decodePathIndex(ix, options.savePathTemplate, da.size());
-                    for (unsigned int i=0; i <= n && i < da.size(); i++) {
+                    int n = decodePathIndex(ix, options.savePathTemplate, da.size());
+                    if (n > 0) {
```

Comment:
I think this should be `n >= 0` for the `%p4`/`%p-1` case shown in the tool-tip.

Reply:
Yikes! In my "testing" on Ubuntu I totally missed the slash disappearing for `%p-1` (note to self: get more sleep). I'm glad you caught it.